src/mono/netcore/System.Private.CoreLib/src/System/Delegate.Mono.cs

```diff
@@ -214,7 +214,7 @@ public static Delegate CreateDelegate(Type type, object? firstArgument, MethodIn
             return CreateDelegate_internal(type, null, info, throwOnBindFailure);
         }
 
-        private static MethodInfo? GetCandidateMethod(RuntimeType type, [DynamicallyAccessedMembers(AllMethods)] Type target, string method, BindingFlags bflags, bool ignoreCase)
+        private static MethodInfo? GetCandidateMethod(RuntimeType type, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type target, string method, BindingFlags bflags, bool ignoreCase)
```

Comment:
Could you add a note why we need `.All` here? I can see someone optimizing this to AllMethods a couple years from now and given the illink deficiency and no trimming test, we might not notice.

Reply:
Good call. I'll both add a comment and a trimming test for this.